ext/random/randomizer.c

```diff
@@ -390,13 +390,13 @@ PHP_METHOD(Random_Randomizer, getBytesFromString)
 	const size_t source_length = ZSTR_LEN(source);
 	const size_t max_offset = source_length - 1;
 
-	if (source_length < 1) {
-		zend_argument_value_error(1, "cannot be empty");
+	if (source_length < 1 && length > 0) {
+		zend_argument_value_error(1, "cannot be empty when argument #2 ($length) is greater than 0");
 		RETURN_THROWS();
 	}
```

Comment:
`source` should never allowed to be empty, even if the requested length is zero. That is an unnecessary special case that is also hard to document.

Reply:
for consistency, i would prefer to allow it, but cba arguing about it (it's an edge-case after all)

removed.